azurelinuxagent/ga/extensionprocessutil.py

```diff
@@ -159,11 +156,7 @@ def read_output(stdout, stderr):
         stderr = ustr(stderr.read(TELEMETRY_MESSAGE_MAX_LEN), encoding='utf-8',
                       errors='backslashreplace')
 
-        def redact(s):
-            # redact query strings that look like SAS tokens
-            return SAS_TOKEN_RE.sub(r'\1<redacted>', s)
-
-        return format_stdout_stderr(redact(stdout), redact(stderr))
+        return format_stdout_stderr(redact_sas_token(stdout), redact_sas_token(stderr))
```

Comment:
event.py instead?

Reply:
This output goes to status file if extension fails with error code, so we still need it here

Reply:
should we then redact in the method that creates the status report?

Reply:
I don't like the idea; we need to do at handler status and extension status etc. But if that's the way, I'll do it in all those places

Reply:
can we do it as part of preparing the report?


```
    def prepare(self, blob_type):
        logger.verbose("Prepare status blob")
        self.data = self.to_json()
        self.type = blob_type

```

Reply:
This will not prevent writing sensitive information to status file. May be sanitizing before we write is correct place

Reply:
what status file?

Reply:
Handlerstatus and extension status file in status folder